smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/AbstractEmitter.java

```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    @Override
+    public synchronized void complete() {
+        verify(internal, name).complete();
+    }
+
+    @Override
+    public synchronized void error(Exception e) {
+        if (e == null) {
+            throw ex.illegalArgumentForException("null");
+        }
+        verify(internal, name).fail(e);
+    }
+
+    @Override
+    public synchronized boolean isCancelled() {
+        MultiEmitter<? super Message<? extends T>> emitter = internal.get();
+        return emitter == null || emitter.isCancelled();
+    }
+
+    @Override
+    public boolean hasRequests() {
+        MultiEmitter<? super Message<? extends T>> emitter = internal.get();
+        return !isCancelled() && emitter.requested() > 0;
+    }
+
+    Multi<Message<? extends T>> getPublisherForStrategy(OnOverflow.Strategy overFlowStrategy, long bufferSize,
+            long defaultBufferSize,
+            Consumer<MultiEmitter<? super Message<? extends T>>> deferred) {
+        switch (overFlowStrategy) {
+            case BUFFER:
+                if (bufferSize > 0) {
+                    return ThrowingEmitter.create(deferred, bufferSize);
+                } else {
+                    return ThrowingEmitter.create(deferred, defaultBufferSize);
+                }
+
+            case UNBOUNDED_BUFFER:
+                return Multi.createFrom().emitter(deferred, BackPressureStrategy.BUFFER);
+
+            case THROW_EXCEPTION:
+                return ThrowingEmitter.create(deferred, 0);
+
+            case DROP:
+                return Multi.createFrom().emitter(deferred, BackPressureStrategy.DROP);
+
+            case FAIL:
+                return Multi.createFrom().emitter(deferred, BackPressureStrategy.ERROR);
+
+            case LATEST:
+                return Multi.createFrom().emitter(deferred, BackPressureStrategy.LATEST);
+
+            case NONE:
+                return Multi.createFrom().emitter(deferred, BackPressureStrategy.IGNORE);
+
+            default:
+                throw ex.illegalArgumentForBackPressure(overFlowStrategy);
+        }
+    }
+
+    /**
+     * Creates the stream when using the default buffer size.
+     *
+     * @param defaultBufferSize the default buffer size
+     * @param stream the upstream
+     * @return the stream.
+     */
+    Multi<Message<? extends T>> getPublisherUsingBufferStrategy(long defaultBufferSize,
+            Multi<Message<? extends T>> stream) {
+        int size = (int) defaultBufferSize;
+        return stream
+                .on().overflow().buffer(size - 2)
+                .onFailure().invoke(t -> synchronousFailure.set(t));
+    }
+
+    public Publisher<Message<? extends T>> getPublisher() {
+        return publisher;
+    }
+
+    boolean isSubscribed() {
+        return internal.get() != null;
+    }
+
+    protected synchronized void emit(Message<? extends T> message) {
+        MultiEmitter<? super Message<? extends T>> emitter = verify(internal, name);
```

Comment:
there is a missing check, probably missing before, `message` must not be `null`.

Reply:
Previously it was done in the `send()` method, which it still is, but can add it here too as extra precaution

Reply:
oh ok, missed it. So it's good.